### DISCUSSION TITLE
How to prevent schema changes in production?

### DISCUSSION THREAD

**Discussion by levino** [pocketbase/pocketbase#5033] [Q&A] [answered]
I hope this question was not answered before. I searched, but could not find this specific topic.

I think it is recommended to run schema migrations in a "file based" manner. So basically do them in a local dev instance of `pocketbase` and then upload the `pb_migrations` folder to the production server and restart `pocketbase` (or maybe it just picks it up by itself?).

What I fear is that someone by accident will edit a table via the admin dashboard of the live instance. I might even do it myself because I simply have selected the wrong browser tab. Is it possible to prevent any changes to the schema via the admin dashboard for a production instance? Would that even be a good idea?

**Comment by ganigeorgiev** [comment, marked as answer]
There is no option to disable collection changes out of the box (at least not without hooks) but you can enable the _Admin UI > Settings > Application > "Hide collection create and edit controls"_ toggle on production to prevent accidental schema changes.

**Reply by levino** [reply]
Thank you @ganigeorgiev. Not only a fantastic product but also a superb maintainer!
